ios/connect.go

```diff
@@ -199,18 +220,26 @@ func ConnectToTunnel(device DeviceEntry, port int) (*net.TCPConn, error) {
 	return conn, nil
 }
 
-func ConnectToHttp2WithAddr(a string, port int) (*http.HttpConnection, error) {
-	addr, err := net.ResolveTCPAddr("tcp6", fmt.Sprintf("[%s]:%d", a, port))
-	if err != nil {
-		return nil, fmt.Errorf("ConnectToHttp2WithAddr: failed to resolve address: %w", err)
-	}
+func ConnectToHttp2WithAddr(a string, port int, userSpaceTUN bool) (*http.HttpConnection, error) {
```

Comment:
having the argument `userSpaceTUN` all over the place doesn't feel nice. We should try to either move this into an interface, or have separate functions with proper names for the two ways of connecting to the device

Reply:
It's probably a bit to much for this PR, but how about adding methods for opening up connections to the device to `DeviceEntry`. It's always a device that we want to connect to and a `DeviceEntry` basically represents that

Reply:
It is something I actually tried yesterday. Either this or a connection module taking a device as a param. I like the DevicrEntry being a simple entity without too much stuff attached

Reply:
That's also fine, assuming that you mean package and not module

Reply:
Yes.